google/cloud/bigquery/job/base.py

```diff
@@ -801,7 +801,7 @@ def reload(
         self,
         client=None,
         retry: "retries.Retry" = DEFAULT_RETRY,
-        timeout: Optional[float] = None,
+        timeout: Optional[Union[float, object]] = POLLING_DEFAULT_VALUE,
```

Comment:
I'd much rather this defaulted to the same timeout as `get_job()` and anywhere that calls `reload()`, we make sure we do the check for `POLLING_DEFAULT_VALUE` before calling `reload()` to make sure we only pass in `float` or `None` or omit the value.

Reply:
done